code_generation/data/attribute_classes/input.json

```diff
@@ -127,6 +127,33 @@
                 }
             ]
         },
+        {
+            "name": "GenericBranchInput",
+            "base": "BranchInput",
+            "attributes": [
+                {    
+                    "data_type": "double",
+                    "names": ["r1","x1","g1","b1"],
+                    "description": "positive sequence parameters"
+                },
+                {
+                    "data_type": "double",
+                    "names": "k",
+                    "description": "transformer ratio, default = 1.0"
```

Comment:
What is the definition of user-provided transformer ratio?

Is it regular transformer ratio or off-nominal ratio? 

For example, given a 155/11 kV transformer connecting to the nodes with rated voltage of 150kV and 10kV.

The regular ratio is: 155/11= 14.09
The off-nominal ratio is: (155/11) / (150/10) = 0.9393

Based on your implementation below, you seem to have chosen for the off-nominal ratio. Please also document this clearly in the attribute description and component documentation.